src/sessions/PageManager.ts

```diff
@@ -28,13 +36,30 @@ export type Attributes = {
  *
  * The interaction level is the order of a page in the sequence of pages sorted by the time they were viewed.
  */
-export class PageManager {
+export class PageManager extends MonkeyPatched {
```

Comment:
suggestion: Create a helper class to handle timing virtual page loads. This way we have separation of concerns by separating the page-interaction state from the page load timing state (increases maintainability).

Reply:
Sounds good, will reflect in next revision.